elastic4s-core/src/main/scala/com/sksamuel/elastic4s/ElasticClient.scala

```diff
@@ -60,12 +63,49 @@ case class ElasticClient(client: HttpClient) extends AutoCloseable {
     }
   }
 
+  private def authenticate(request: ElasticRequest, authentication: Authentication): ElasticRequest = {
+    authentication match {
+      case Authentication.UsernamePassword(username, password) =>
+        request.addHeader(
+          "Authorization",
+          "Basic " + Base64.getEncoder.encodeToString(s"$username:$password".getBytes)
+        )
+      case Authentication.ApiKey(apiKey) =>
+        request.addHeader(
+          "Authorization",
+          "ApiKey " + Base64.getEncoder.encodeToString(apiKey.getBytes)
+        )
+      case Authentication.NoAuth =>
+        request
+    }
+  }
+
 
   def close(): Unit = client.close()
 }
 
-case class CommonRequestOptions(timeout: Duration, masterNodeTimeout: Duration, headers: Map[String, String] = Map.empty)
+sealed trait Authentication
+
+object Authentication {
+  case class UsernamePassword(username: String, password: String) extends Authentication
+
+  case class ApiKey(apiKey: String) extends Authentication
+
+  case object NoAuth extends Authentication
+}
+
+case class CommonRequestOptions(
+  timeout: Duration,
+  masterNodeTimeout: Duration,
+  authentication: Authentication,
+  headers: Map[String, String] = Map.empty
```

Comment:
Can you switch these around, for slightly better backwards compatibility?

Reply:
Fixed, and gave it default value